### ISSUE TITLE
Plots in AnalyzeHyperoptResults

### ISSUE THREAD

**Issue by mpariente** [guillaume-chevalier/Hyperopt-Keras-CNN-CIFAR-100#1]
Hi, 
When you analyze the results in the notebook, you plot `accuracy = [1.0/100] + neural_net["history"]["fine_outputs_acc"]` and call it the test accuracy however, it seems to me that it is the training accuracy. 
I think you should plot `accuracy = [1.0/100] + neural_net["history"]["val_fine_outputs_acc"]` to obtain the test accuracy.  

Did I miss something? Are you more interested in the training accuracy in fucntion of the hyper-parameters than the test (or validation) accuracy? 

Thanks in advance 

Edit : I would have liked to creat the issue in the Vooban repo but I can't..

**Comment by guillaume-chevalier** [comment]
Hi, 

From what I see, you are talking about the `AnalyzeHyperoptResults.ipynb` file here: 
https://github.com/Vooban/Hyperopt-Keras-CNN-CIFAR-100/blob/master/AnalyzeHyperoptResults.ipynb

It is truly an error that should be corrected, thanks for reporting that. At least, when I report the best results in the `README.md`, I am truly talking about the test accuracy which is `max_acc = max(history['val_fine_outputs_acc'])`. I'll fix the `AnalyzeHyperoptResults.ipynb` file later on.

**Comment by mpariente** [comment]
Hi, 

Indeed I was talking about this. I'm looking forward to see the graphs for the validation error (I don't have a killer GPU yet..)

One a related subject, do you have any idea why for some nets, the training accuracy drops to zero after a certain number of epochs? It would seem ok if it would be the test accuracy (however quite drastic) but for the training accuracy? 
It seems it only happens with `RMSProp` and `elu` activation for small L2 regularization coefficients.

By the way, I love your implementation, it's much more convenient than hyperas because it can be used to hyper-optimize preprocessing and data-augmentation, which is really great, thank you !

**Comment by guillaume-chevalier** [comment]
RMSProp is known to be a bit unstable. I'd guess that at some point there are some NaN values in the neural net for a reason I haven't investigated yet, like if the gradient contained a NaN at some point or a division by zero (during batch norm maybe?).

Also, if it only happens with the ELU, this is another thing that wouldn't surprise me, because the ELU needs to compute an exponential, which can have more numerical errors than a simple ReLU which is just simply `max(0, x)` without tampering with the values.
